File: pset2/caesar/caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>
#include <cs50.h>

int main(int argc, string argv[])
{
    // Checks if command-line arguments was used correctly
    if (argc == 2)
    {
        // Converts the command-line argumento from string to a int
        int k = atoi(argv[1]);

        // Prompts user for text
        string s = get_string("plaintext: ");

        // Loop to go through the string
        for (int i = 0; i < strlen(s); i++)
        {
            // Checks if it is alphabetic upper or lower and cipher it
            if (isalpha(s[i]))
            {
                if (islower(s[i]))
                {
                    s[i] = 'a' + ((s[i] - 'a') + k) % 26;
                }
                else
                {
                    s[i] = 'A' + ((s[i] - 'A') + k) % 26;
                }
            }

        }
        printf("ciphertext: %s\n", s);
    }
    else
    {
        // Prints error message if command-line argument was used incorrectly
        printf("Error\n");
        return 1;
    }
}

File: cs50_edx/pset6/caesar/caesar.py
import sys
from cs50 import get_string

# Checks for correct arguments usage
if len(sys.argv) != 2:
    sys.exit("Usage: python caesar.py n")

# Prompts user for text
s = get_string("plaintext: ")

# Convert input argument into an integer
k = int(sys.argv[1])

print("ciphertext: ", end="")

# Ciphers the text iterating through the chars of the text string
for c in s:
    if c.isalpha():
        if c.isupper():
            c = chr(((ord(c) - ord('A')) + k) % 26 + ord('A'))
        else:
            c = chr(((ord(c) - ord('a')) + k) % 26 + ord('a'))
    print(c, end="")
print()
